### PR TITLE
Remove is_stable from unit group and script

### PR DESCRIPTION
**Pull request by dmcavoy**
Removes is_stable property from unit_group and script and uses the published_state instead.

Many of the files in this PR are script, script_json, or course files where I'm removing the is_stable setting.

## Links

https://codedotorg.atlassian.net/browse/PLAT-1059

## Testing story

- Relying on existing tests and updates to them

### CODE REVIEW

**Comment by dmcavoy** [comment]
I think the test failures on this PR will be resolved once the course, script and script_json files update from the migration in https://github.com/code-dot-org/code-dot-org/pull/41026.

**Comment by dmcavoy** [comment]
I want to look at some of the courses without family names to make sure they are still showing up in the dropdown. Just leaving here incase I don't get to it

**Comment by davidsbailey** [comment]
top-level comment -- it seems like this will break https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20210617011456_set_published_state_attempt_two.rb#L1 for any developers or adhocs who have not run it already. I would suggest simply nulling out that migration as part of this PR. in hindsight, the migration didn't really need to run in every environment -- because the published_state is now being set in .script and .script_json files, it will be seeded into every environment even without this migration.

**Comment by dmcavoy** [comment]
> top-level comment -- it seems like this will break https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20210617011456_set_published_state_attempt_two.rb#L1 for any developers or adhocs who have not run it already. I would suggest simply nulling out that migration as part of this PR. in hindsight, the migration didn't really need to run in every environment -- because the published_state is now being set in .script and .script_json files, it will be seeded into every environment even without this migration.

Great I will null out the migration here and remove get_published_state

File: dashboard/app/models/unit_group.rb [hunk 1]
```diff
@@ -535,7 +534,7 @@ def self.latest_stable_version(family_name)
       # select only courses in the same course family.
       where("properties -> '$.family_name' = ?", family_name).
       # select only stable courses.
-      where("properties -> '$.is_stable'").
+      where(published_state: SharedConstants::PUBLISHED_STATE.stable).
```
Comment (davidsbailey) [inline dashboard/app/models/unit_group.rb:522]: 👍

**Comment by dmcavoy** [comment]
> > top-level comment -- it seems like this will break https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/db/migrate/20210617011456_set_published_state_attempt_two.rb#L1 for any developers or adhocs who have not run it already. I would suggest simply nulling out that migration as part of this PR. in hindsight, the migration didn't really need to run in every environment -- because the published_state is now being set in .script and .script_json files, it will be seeded into every environment even without this migration.
> 
> Great I will null out the migration here and remove get_published_state

Actually, I'm gonna keep the part of the migration that adds default value and disallow null.

**Review by davidsbailey — APPROVED** [review]
Great work Dani! This is nice and clean with the `stable?` helper method and giving this change its own PR.

**Pull request merged**